DXMainClient/DXGUI/Multiplayer/GameLobby/GameLobbyBase.cs

```diff
@@ -535,28 +535,46 @@ protected void ListMaps()
             lbGameModeMapList.SelectedIndex = -1;
 
             int mapIndex = -1;
-            int skippedMapsCount = 0;
 
             var isFavoriteMapsSelected = IsFavoriteMapsSelected();
             var maps = GetSortedGameModeMaps();
 
             bool gameModeMapChanged = false;
 
-            for (int i = 0; i < maps.Count; i++)
+            List<GameModeMap> filteredMaps;
+
+            if (tbMapSearch.Text != tbMapSearch.Suggestion)
             {
-                var gameModeMap = maps[i];
-                if (tbMapSearch.Text != tbMapSearch.Suggestion)
+                var search = tbMapSearch.Text.ToUpperInvariant();
```

Comment:
I am thinking that maybe it's better to use `stringA.Contains(stringB, IrrelevantCultureIgnoreCase)` for untranslated name and `CurrentCultureIgnoreCase` for (translated) Name, instead of making them upper cases.

This might be a nitpick so the choice is yours.